drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1565,6 +1565,11 @@ int usb_dc_ep_clear_stall(const uint8_t ep)
 		return -EINVAL;
 	}
 
+	if (NRF_USBD_EPISO_CHECK(ep)) {
+		/* ISO transactions do not support a handshake phase. */
+		return -EIO;
```

Comment:
I think that `-EINVAL` would be a more appropriate error code for this case. `-EIO` suggests that some operation involving hardware failed, and here the request is simply wrong.